tt_metal/hw/inc/ethernet/dataflow_api.h

```diff
@@ -50,14 +70,44 @@ volatile uint32_t *RtosTable =
     (volatile uint32_t *)&__erisc_jump_table;  // Rtos Jump Table. Runtime application needs rtos function handles.;
 
 FORCE_INLINE
-void reset_erisc_info() { erisc_info->user_buffer_bytes_sent = 0; }
+void reset_erisc_info() {
+    erisc_info->per_channel_user_bytes_send[0].bytes_sent = 0;
+    erisc_info->per_channel_user_bytes_send[0].receiver_ack = 0;
+    erisc_info->per_channel_user_bytes_send[1].bytes_sent = 0;
+    erisc_info->per_channel_user_bytes_send[1].receiver_ack = 0;
+    erisc_info->per_channel_user_bytes_send[2].bytes_sent = 0;
+    erisc_info->per_channel_user_bytes_send[2].receiver_ack = 0;
+    erisc_info->per_channel_user_bytes_send[3].bytes_sent = 0;
+    erisc_info->per_channel_user_bytes_send[3].receiver_ack = 0;
+    erisc_info->per_channel_user_bytes_send[4].bytes_sent = 0;
+    erisc_info->per_channel_user_bytes_send[4].receiver_ack = 0;
+    erisc_info->per_channel_user_bytes_send[5].bytes_sent = 0;
+    erisc_info->per_channel_user_bytes_send[5].receiver_ack = 0;
+    erisc_info->per_channel_user_bytes_send[6].bytes_sent = 0;
+    erisc_info->per_channel_user_bytes_send[6].receiver_ack = 0;
+    erisc_info->per_channel_user_bytes_send[7].bytes_sent = 0;
+    erisc_info->per_channel_user_bytes_send[7].receiver_ack = 0;
+}
 
 namespace internal_ {
 
+/**
+ *
+ * Return value: None
+ *
+ * | Argument          | Description                                             | Type     | Valid Range | Required |
+ * |-------------------|---------------------------------------------------------|----------|-------------|----------|
+ * | num_words         | number of eth link words (16B)                          | uint32_t | 0..1MB      | True     |
+ */
 FORCE_INLINE
 void eth_send_packet(uint32_t q_num, uint32_t src_word_addr, uint32_t dest_word_addr, uint32_t num_words) {
+    uint32_t count = 0;
     while (eth_txq_reg_read(q_num, ETH_TXQ_CMD) != 0) {
-        risc_context_switch();
+        if (count > 1000000) {
```

Comment:
Is this debug? or are you checking this in? we can put an overall timing check into the implementation of risc_context_switch

Reply:
I can omit this but it does make a noticeable difference for perf (as in gigabytes per second lost without it). The value itself can probably be tweaked down considerably while addressing both needs. I would say anytime we are checking the tx queue status, it's probably worth it to have a busy loop before context switch because in most cases, the reg will clear quickly. 

That being said, maybe the best course of action is for me to omit this for now and open an issue to track it and tweak it locally during development.